### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by agarciamontoro**
#### Summary
Documentation is a good thing to do when you don't have any connection to the outside world. No rush at all in reviewing this, but shouldn't take too long either to do so :)

- Main changes:
  - The docs for config files were all moved to `docs/config` and renamed to remove the `_config` suffix.
  - The `docs/` directory now has a `README.md` file that serves as intro to all other docs. Hopefully this improves discoverability.
  - The guide for generating data has drastically changed.
  - The setups file is removed. I didn't know where to reference it from the intro document, and it is outdated, so I thought it best to just get rid of it. I can change my mind here if you have a strong opinion, though.
- Minor changes:
  - Typos
  - Promote/demote sections to better structure the content
  - A new FAQ entry

#### Ticket Link
--

### CODE REVIEW

File: docs/readme.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# Documentation
+
+## Where do I start?
+
+The documentation here is comprehensive, so it may be hard for someone new to know where to start reading to understand how the tool works or how it is architected. The best starting point depends on what your end goal is:
+
+- If you are interested in using the tool to run some tests, there are a few documents you need to read, roughly in the following order:
+    - [How-to guide](load-test-how-to-use.md): this guide is a great overview of how things work, as well as tips and tricks that will make your life easier. It also links to multiple other documents, which you can follow to broaden your understanding. Read it completely first before running any tests!
+    - [How to run a load-test locally](local_loadtest.md): a specific guide to start running load-tests locally in your computer, no AWS needed.
+    - [How to run a load-test in Terraform](terraform_loadtest.md): a more advanced guide to deploy a cluster to AWS to run a more powerful load-test.
```
Comment (streamer45) [inline docs/readme.md:10]: Can we add a note that this is the recommended way to do any serious load-test aiming at production?

**Review by streamer45 — APPROVED** [review]
Very thorough, I like it :100:

File: docs/generating_data.md [hunk 2]
```diff
@@ -1,46 +1,119 @@
...
+
+With everything setup and configured, we get to the main step: generating the data.
+
+There are two more variables that you need to choose:
+- The number of simulated users that will run the actions to generate the data.
+- The rate of each simulated user; i.e., the speed at which it will run the actions.
+
+There is no golden rule on how to choose these numbers. The best way to decide is to run a couple of initial tests and monitor  how the system behaves.
+- The more users you run, the faster the data will be generated.
+- The larger the rate is, the slower the data will be generated (yes, this rate is more of a period, not a rate, we'll change the name at some point).
```
[the remark this answers is not in the document]
Comment (agnivade) [reply docs/generating_data.md:63]: It is a period-like multiplier!

**Review by agnivade — APPROVED** [review]
Wow very nice. Nothing else to add apart from Claudio's suggestions!

File: docs/faq.md [hunk 3]
```diff
@@ -25,6 +25,10 @@ To configure this, you need to take a look at the `MonitorConfig.Queries` config
 
 Note that in both cases, `ClusterConfig.MaxActiveUsers` should be set to `AgentInstanceCount * UsersConfiguration.MaxActiveUsers`.
 
+### How many users does an agent support?
+
+For an agent running in a `c5.xlarge` instance in AWS (X CPU, Y RAM), 2000 users is the sweet spot.
```
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply docs/faq.md]: Solved in https://github.com/mattermost/mattermost-load-test-ng/pull/612/commits/88065bf7ca9336e56781bfa885904d4e57b5ed63

File: docs/generating_data.md [hunk 4]
```diff
@@ -1,46 +1,119 @@
...
+2. Then, verify that the [`UserController` in `config/config.json`](config/config.md#usercontrollerconfiguration) is set to `generative`. Also, set all absolute numbers under [`InstanceConfiguration` in that same file](config/config.md#usercontrollerconfiguration) to 0, since some commands do run an `init` process that populates the database with the data configured there. Since we are generating the data ourselves, we don't want anything done for us in that phase.
+3. Then, create or modify the gencontroller configuration file in `config/gencontroller.json`. You can use the sample file as a starting point:
 ```sh
 cp config/gencontroller.sample.json config/gencontroller.json
 ```
+4. Now tweak everything you need in the controller configuration. All the settings are [documented](config/gencontroller.md), but here are some recommendations:
+  - Make sure that the number of DMs is possible with the number of users you will run (more on simulated users later). If you run 10 users, the maximum number of different DMs is 45 so don't use a number larger than that. The formula for the maximum number of DMs is $n(n-1)2$, with $n$ the number of users.
```
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply docs/generating_data.md]: Ah, great catch! Solved in https://github.com/mattermost/mattermost-load-test-ng/pull/612/commits/9094fc56659ba38851d5c854c87c79c44378f833

File: docs/generating_data.md [hunk 5]
```diff
@@ -1,46 +1,119 @@
...
+  - All the settings follow specific distributions in real servers, so make sure to run some numbers in the server you want to mimic to make an educated decision here. As an example, the vast majority of channels in a normal server are DMs and GMs, accounting for 75% and 15% respectively, while private channels represent a 7% and public ones only a 3%.
+  
+You don't need to modify any of the other configuration files, since they are not considered when using a gencontroller.
```
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply docs/generating_data.md:50]: Will address this in a future PR.

File: docs/generating_data.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (agarciamontoro) [reply docs/generating_data.md:63]: Will address this in a future PR. Tracked by https://mattermost.atlassian.net/browse/MM-53684

File: docs/readme.md (hunk 1 above)
Comment (agarciamontoro) [reply docs/readme.md]: Done in https://github.com/mattermost/mattermost-load-test-ng/pull/612/commits/35e6b2f4640b5c000b0e32def7f5439305594278

**Pull request merged**